### ISSUE TITLE
Preferences in private mode doesn't get saved on Firefox.

### ISSUE THREAD

**Issue by syrupderg** [Authenticator-Extension/Authenticator#1149]
### Describe the issue

Authenticator doesn't applies the changes made to private mode, this wasn't a issue on Chromium based or other browsers.

### Browser

Firefox

### Browser Version

122.0

### Extension Version

6.3.5

**Comment by Sneezry** [comment]
See https://github.com/Authenticator-Extension/Authenticator/issues/1085#issuecomment-1731911650

**Comment by syrupderg** [comment]
> See [#1085 (comment)](https://github.com/Authenticator-Extension/Authenticator/issues/1085#issuecomment-1731911650)

i forgot that i made that issue before but it is definitely a bug, other addons doesnt loses preferences but authenticator loses theme and and some of the other preferences, while still keeping the auth code and password that has been set. it works normal on normal tabs but on private tabs it uses the default preferences.

**Comment by Sneezry** [comment]
Could you share some examples of other add-ons that manage to retain user preferences effectively? It would be really interesting to understand how they handle the storage of configuration settings, especially if they're not using local storage. This could give us some great insights.

**Comment by mymindstorm** [comment]
@Sneezry they likely use `chrome.storage.*`. We use `localstorage` for user preferences and `chrome.storage` for accounts.

**Comment by Sneezry** [comment]
@mymindstorm We are transitioning from `localStorage` to `chrome.storage.local` in MV3, which will effectively resolve the issue.

**Comment by mymindstorm** [comment]
@Sneezry Could we add a version field to the storage to make future migrations easier? If that isn't already there.

**Comment by Sneezry** [comment]
@mymindstorm Indeed, that's a logical approach. Let's design a schema for the data stored in the storage system, incorporating a version element and flexible attributes to accommodate future expansions.
